### PR TITLE
feat: localize videos

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
This adds support for embedding

- Language specific youtube videos
- Bilibili videos in the Chinese sites

The bilibili embed iframe comes with three ids (`aid`, `bvid` and `cid`).  It doesn't seem to be necessary to use them all, but I don't really understand what they are ids *for* so I'm hesitant to remove them.

@miyaliu666 @RafaelDavisH could I get some examples of translated videos that I could use?  I need one bilibili video and (ideally) one video for Italian, Spanish and Portuguese if they exist!

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
I added some embed ids so that it can be tested locally.

File: client/src/templates/Challenges/components/VideoPlayer.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+// TODO: pull these types from all-langs
+const { clientLocale } = envData as {
+  clientLocale:
+    | 'english'
+    | 'chinese'
+    | 'chinese-traditional'
+    | 'espanol'
+    | 'italian'
+    | 'portuguese';
```
Comment (nhcarrigan) [inline client/src/templates/Challenges/components/VideoPlayer.tsx:16]: We use clientLocale in a few places - should we build this as an exported type declaration?

Comment (ojeytonwilliams) [reply client/src/templates/Challenges/components/VideoPlayer.tsx:16]: Sounds good, yeah.  After https://github.com/freeCodeCamp/freeCodeCamp/pull/42862 gets in?

Comment (miyaliu666) [reply client/src/templates/Challenges/components/VideoPlayer.tsx:16]: Sorry, I just saw this issue. Here is a translated [bilibili video](https://www.bilibili.com/video/BV1Fh411z7tr) you can use to test. @ojeytonwilliams

Comment (ojeytonwilliams) [reply client/src/templates/Challenges/components/VideoPlayer.tsx:16]: Thanks Miya, I'll put that in before we merge!

**Comment by RafaelDavisH** [comment]
@ojeytonwilliams Sorry to respond after 27 days. For Spanish, we don't have any video translated yet.

**Comment by ojeytonwilliams** [comment]
No worries @RafaelDavisH this can be QA'd as is and, if need be, we can use the original English video as a placeholder for the translated vidoes.

**Comment by ojeytonwilliams** [comment]
QA note: to see the videos locally, you need to set the CLIENT_LOCALE, only.  Leave CURRICULUM_LOCALE=english or it won't build properly.

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/intermediate-strings.md [hunk 2]
```diff
@@ -3,6 +3,10 @@ id: 5e7b9f070b6c005b0e76f061
 title: Intermediate Strings
 challengeType: 11
 videoId: KgT_fYLXnyk
+videoLocaleIds:
+  espanol: 7jjoyy7_RCk
+  italian: 44LA7i_f4ug
+  portuguese: dQw4w9WgXcQ
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/intermediate-strings.md:9]: Ha..ha.. very funny :bemused:

I am not sure I can approve this:
image

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/intermediate-expressions.md [hunk 3]
```diff
@@ -3,6 +3,10 @@ id: 5e7b9f050b6c005b0e76f057
 title: Intermediate Expressions
 challengeType: 11
 videoId: dKgUaIa5ATg
+bilibiliIds:
```
Comment (ahmadabdolsaheb) [inline curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/intermediate-expressions.md:6]: are you imbedding this wrong?
Screen Shot 2021-09-20 at 11 53 53 AM

File: client/src/templates/Challenges/components/VideoPlayer.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                videoId +
+                '&lang=en&bl=watch&ui=hd&ref=wt&tab=captions'
+              }
+              rel='noopener noreferrer'
+              target='_blank'
+            >
+              {t('learn.add-subtitles')}
+            </a>
+            .
+          </i>
```
Comment (ahmadabdolsaheb) [inline client/src/templates/Challenges/components/VideoPlayer.tsx:90]: Might be missing something but this does not show up. Where should this be displayed?

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/intermediate-expressions.md (hunk 3 above)
Comment (ojeytonwilliams) [reply curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/intermediate-expressions.md:6]: I don't think so, no.  I think that's just how they roll, as it were.

File: client/src/templates/Challenges/components/VideoPlayer.tsx (hunk 4 above)
Comment (ojeytonwilliams) [reply client/src/templates/Challenges/components/VideoPlayer.tsx]: Thanks Ahmad, I've re-removed them.

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/intermediate-expressions.md (hunk 3 above)
Comment (miyaliu666) [reply curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/intermediate-expressions.md:6]: The flying words are bilibili bullet screen, which allows real-time comments from viewers to fly across the screen like bullets.

**Comment by nhcarrigan** [comment]
We will need to revert these rick roll videos pre-merge right?

File: client/src/templates/Challenges/video/Show.tsx [hunk 5]
```diff
@@ -322,6 +315,16 @@ export const query = graphql`
   query VideoChallenge($slug: String!) {
     challengeNode(fields: { slug: { eq: $slug } }) {
       videoId
+      videoLocaleIds {
+        espanol
+        italian
+        portuguese
+      }
```
Comment (nhcarrigan) [inline client/src/templates/Challenges/video/Show.tsx:322]: To confirm, as we roll out new languages we'll also need to add them here?

**Comment by ojeytonwilliams** [comment]
> We will need to revert these rick roll videos pre-merge right?

Of course. There needs to be at least one video for each language, but that's it.

File: client/src/templates/Challenges/video/Show.tsx (hunk 5 above)
Comment (ojeytonwilliams) [reply client/src/templates/Challenges/video/Show.tsx:322]: Right. Anything we want a localised video for we'd need to add to this query.

**Review by naomi-lgbt — DISMISSED** [review]

**Comment by nhcarrigan** [comment]
I'm good with this pending the revert of the rick roll

File: client/src/templates/Challenges/video/Show.tsx (hunk 5 above)
Comment (nhcarrigan) [reply client/src/templates/Challenges/video/Show.tsx:322]: Made a PR to add this to the docs.

**Review by naomi-lgbt — APPROVED** [review]

**Review by ahmaxed — APPROVED** [review]

**Pull request merged**